### PR TITLE
header name constants changed from `&str` to `HeaderName`

### PR DESCRIPTION
**Pull request by ctaggart**
- header name constants changed from `&str` to `azure_core::headers::HeaderName`
- replaced types from `http:header` with `azure_core::headers` types
- create `const fn` `HeaderName::from_static`
- `Header::get` takes a `&HeaderName` instead of `Into<HeaderName>`
- similar changes for `HeaderValue`
- removed `Request::headers_mut`

### CODE REVIEW

File: sdk/storage_datalake/tests/files.rs [hunk 1]
```diff
@@ -77,7 +77,8 @@ async fn file_upload() -> azure_core::Result<()> {
     Ok(())
 }
 
-#[tokio::test]
+// #[tokio::test]
+#[allow(dead_code)]
```
Comment (cataggar) [inline sdk/storage_datalake/tests/files.rs:81]: I need help figuring out why this one mock test for storage_datalake is failing.

File: sdk/core/src/request_options/content_language.rs [hunk 2]
```diff
@@ -33,7 +33,7 @@ impl From<&String> for ContentLanguage {
 
 impl Header for ContentLanguage {
     fn name(&self) -> headers::HeaderName {
-        http::header::CONTENT_TYPE.into()
```
Comment (cataggar) [inline sdk/core/src/request_options/content_language.rs:36]: Was this a bug in a recent commit?

File: sdk/storage_datalake/tests/files.rs (hunk 1 above)
Comment (cataggar) [reply sdk/storage_datalake/tests/files.rs:81]: > received request have header 'range' but it was not present in the read request

File: sdk/core/src/request_options/range.rs [hunk 3]
```diff
@@ -28,9 +28,12 @@ impl AsHeaders for Range {
     type Iter = std::vec::IntoIter<(HeaderName, HeaderValue)>;
 
     fn as_headers(&self) -> Self::Iter {
-        let mut headers = vec![("x-ms-range".into(), format!("{}", self).into())];
+        let mut headers = vec![(headers::RANGE, format!("{}", self).into())];
```
Comment (cataggar) [inline sdk/core/src/request_options/range.rs:31]: should be MS_RANGE

Comment (cataggar) [reply sdk/core/src/request_options/range.rs]: This was the bug caught by the test.

**Review by bmc-msft — APPROVED** [review]

**Pull request merged**